### ISSUE TITLE
Fix Request: AvoidMultipleConcatStatements spurious violations and missing cases

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#434]

**Comment by jborgers** [comment]
```
class Issue435 {

    private String[] values = {"tic", "tac", "toe"};

    String bad1() {
        String logStatement = "";
        logStatement += values[0];
        logStatement = logStatement + values[1]; // bad <-- missed case
        return logStatement; 
    }

    String bad2() {
        String logStatement = "";
        logStatement = logStatement + values[1];
        logStatement += values[0]; // bad <-- missed case
        return logStatement; 
    }

    String bad3() {
        String logStatement = "";
        logStatement = logStatement + values[0];
        logStatement += values[1] + "-"; // bad
        logStatement += values[2]; // <-- 'false positive', spurious
        logStatement += "-bla"; // <-- 'false positive', spurious
        return logStatement; // bad
    }

    String bad4() {
        String logStatement = "";
        logStatement = logStatement + values[0];
        logStatement = logStatement + values[1] + "-"; // bad
        logStatement = logStatement + values[2]; // <-- 'false positive', spurious
        logStatement = logStatement + "-bla"; // <-- 'false positive', spurious
        return logStatement; // bad
    }
}

```
